### ISSUE TITLE
Can't collect original anti-cube if Ach. Cube is collected

### ISSUE THREAD

**Issue by MistahKurtz** [renaudbedard/fez-1.12-issues#23]
Setup: Go into a room where a code can be executed to get an anticube. Put in the code for the room's anticube. Put in the achievement cube code. Collect ach. cube. Leave room without collecting the room's anticube. Re-enter room. Input code for room's anticube. Room's anticube does not spawn.

(By "room", I mean any specific area. This affects anticube pillars' anticubes.)

**Comment by MistahKurtz** [comment]
Tuning fork anticubes are unaffected.

**Comment by 0x0ade** [comment]
From what I've seen, tuning fork anticubes aren't affected as they aren't handled as level scripts, but the engine.

Taking a look at an excerpt of the level ZU_BRIDGE, XML-ified...

```
    <Entry key="5">
      <Script name="Untitled" oneTime="False" triggerless="False" ignoreEndTriggers="False" levelWideOneTime="False" disabled="False" isWinCondition="True">
        <Triggers>
          <ScriptTrigger event="CollectedAnti">
            <Entity entityType="Gomez"/>
          </ScriptTrigger>
        </Triggers>
        <Conditions>
          <ScriptCondition property="GetLevelState" value="NOT_COLLECTED" operator="Equal">
            <Entity entityType="Game"/>
          </ScriptCondition>
        </Conditions>
        <Actions>
          <ScriptAction operation="SetLevelState" killswitch="False" blocking="False">
            <Entity entityType="Game"/>
            <Arguments>
              <Entry></Entry>
            </Arguments>
          </ScriptAction>
        </Actions>
      </Script>
    </Entry>
```

... shows that the script detects any anticube collected and thinks that any anticube is the level anticube. Ouch :/

This may sound horrible, but maybe assign anticubes an ID that the scripts then can access, with cross-level anticubes having a negative (thus completely different) ID? This may also fix other duplicate anticubes (the achievement-qr-dupe if that's still a thing), but is quite some undertaking for FEZ.

I guess Renaud has got a better idea.

**Comment by renaudbedard** [comment]
Aye... yeah, this is a tricky one. It affects 14 different levels total.

I'll try to think of a clean way to fix this, but the current way cubes are respawned-if-not-collected is really easy to break because of the Achievement anticube, and because of the QR map anticube.

A tagging system for the cubes to respawn is a good idea, but not trivial to implement. Tagging the achivement+map cubes would be easier than modifying all the level scripting (which is super risky for creating new bugs).

I'll come back to it!

**Comment by renaudbedard** [comment]
Found a good solution for this bug! Similar to what I (and Maik) suggested earlier.
Basically, the achievement and map-QR anti-cubes are marked as special, and don't cause the _CollectedAnti_ script event to be raised. Instead, they do they logic using another event.
So when collecting them, the scripts don't know that an anti-cube has been collected and all is well.

I tested an FPS post and a QR code, but the logic is the same everywhere. (in all 14 levels : `BOILEROOM`, `CODE_MACHINE`, `LIGHTHOUSE`, `NUZU_SCHOOL`, `PARLOR`, `TELESCOPE`, `TREE`, `TREE_SKY`, `WATERTOWER_SECRET`, `ZU_BRIDGE`, `ZU_CODE_LOOP`, `ZU_HOUSE_EMPTY`, `ZU_HOUSE_QR` and `ZU_THRONE_RUINS` -- feel free to test! :D)
